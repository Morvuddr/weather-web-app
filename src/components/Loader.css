.Loader{
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}
h1 {
    text-align: center;
}
.Spinner{
    border: 16px solid #f3f3f3;
    border-top: 16px solid gray;
    border-radius: 50%;
    width: 60px;
    height: 60px;
    animation: spin 2s linear infinite;
}

@keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
}