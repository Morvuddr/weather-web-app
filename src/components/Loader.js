import React from 'react';
import './Loader.css';

export function Loader(){
    return(
        <div className="Loader">
            <h1>Подождите, данные загружаются</h1>
            <div className="Spinner">

            </div>
        </div>
    );
}