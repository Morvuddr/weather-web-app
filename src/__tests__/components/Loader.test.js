import Loader from '../../components/Loader';
import renderer from 'react-test-renderer';
import React from 'react';

describe('Loader component', () => {
    test('matches the snapshot', () => {
        const tree = renderer.create(
            <Loader />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});