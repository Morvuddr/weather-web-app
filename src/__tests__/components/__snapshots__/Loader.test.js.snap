// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component matches the snapshot 1`] = `
<div
  className="Loader"
>
  <h1>
    Подождите, данные загружаются
  </h1>
  <div
    className="Spinner"
  />
</div>
`;
